windows-driver-docs-pr/develop/removing-coinstallers.md

```diff
@@ -0,0 +1,81 @@
+### INF File
+A driver package INF can set the device friendly name as follows:
+```
+[DDInstall.HW]
+AddReg = FriendlyName_AddReg
+
+[FriendlyName_AddReg]
+HKR,,FriendlyName,, "Device Friendly Name"
+```
+
+### Runtime
+The friendly name can be set by the driver during the start IRP or the PrepareHardware phase by setting the DEVPKEY_Device_FriendlyName property with one of the following APIs:  
+[IoSetDevicePropertyData](../drivers/ddi/wdm/nf-wdm-iosetdevicepropertydata.md)  
+[WdfAssignProperty](../ddi/wdfdevice/nf-wdfdevice-wdfdeviceassignproperty.md)
+
+## Other device settings/configurations:
+When possible, the driver can change the device settings and configuration within the driver Start IRP or the PrepareHardware phase. When modifying state at runtime, the driver should follow the [driver package isolation requirements](./driver-isolation.md). These requirements contain the guidance on driver configuration and state layout and help future-proof the driver by making it more resilient to external changes, easier to update, and more straightforward to install.
+  
+For settings and configuration that cannot be set within the driver itself, a driver package may also include user-mode runtime components that alter the settings and configuration. This can be a user-facing app or a Win32 service that updates the configuration. If a persistent component such as a service is used, ensure that its functionality is necessary and cannot be performed in a less resource-intensive manner, such as within a driver package INF or within the driver itself. Great care should be taken to ensure that any service is only running when the relevant devices are connected (see [Service Triggers](../../windows/win32/services/service-trigger-events.md), [Win32 services interacting with devices](../install/best-practices-win32services-interacting-with-devices.md), and [Registering for device interface notifications](../drivers/install/registering-for-notification-of-device-interface-arrival-and-device-removal.md)), and the service meets the latest requirements (for instance, passing [API Validator](./validating-windows-drivers.md#apivalidator)).
+
+## Installing Associated Software
+The 'Componentized' portion of the [DCH driver requirements](../develop/dch-principles-best-practices.md) introduced a concept called the SoftwareComponent, which is a mechanism to decouple the install of a device driver from it associated software. When a software component is created by the INF, it will automatically create a child device that maps to the software component. This child device will exist for the purpose of installing the software associated to the parent device. This software can be installed and updated independently of the main device and driver.
+
+Within a SoftwareComponent driver package INF, the recommended mechanism to install software is using an AddSoftware directive. This will trigger the download an installation of software from the Windows Store.  
```

Comment:
```suggestion
Within a SoftwareComponent driver package INF, the recommended mechanism to install software is using an AddSoftware directive. This will trigger the download and installation of software from the Windows Store.  
```